src/test/fs_tests.cpp

```diff
@@ -52,6 +52,25 @@ BOOST_AUTO_TEST_CASE(fsbridge_fstream)
         file >> input_buffer;
         BOOST_CHECK_EQUAL(input_buffer, "bitcoin");
     }
+    {
+        // Join an absolute path and a relative path.
+        fs::path p = fsbridge::AbsPathJoin(tmpfolder, "fs_tests_₿_🏃");
+        BOOST_CHECK(p.is_absolute());
+        BOOST_CHECK_EQUAL(tmpfile1, p);
+    }
+    {
+        // Join two absolute paths.
+        fs::path p = fsbridge::AbsPathJoin(tmpfile1, tmpfile2);
+        BOOST_CHECK(p.is_absolute());
+        BOOST_CHECK_EQUAL(tmpfile2, p);
+    }
+    {
+        // Joining with empty path behaves differently in various filesystem implementations.
+        fs::path p = fsbridge::AbsPathJoin(tmpfolder, "");
+        BOOST_CHECK(p.is_absolute());
+        // "tmpfolder" does not contain trailing slash. No trailing slash should be added by the tested method.
+        BOOST_CHECK(p.string().back() != '/');
```

Comment:
In commit "Replace fs::absolute calls with AbsPathJoin calls" (215c78465059175505218566f2ec38d52d3be827)

I don't think the comment and test are quite right. Joining behavior is well defined for boost and std path types and does not depend on the filesystem. The differences are between the boost library behavior and the standard library behavior are design choices, not OS differences. Would suggest replacing with:

```c++
{
     // Ensure joining with empty paths does not add trailing path components.
     BOOST_CHECK_EQUAL(tmpfile1, fsbridge::AbsPathJoin(tmpfile1, ""));
     BOOST_CHECK_EQUAL(tmpfile1, fsbridge::AbsPathJoin(tmpfile1, {}));
}
```

This is the empty path behavior that we are relying on AbsPathJoin and boost::filesystem::absolute functions for that the C++17 standard library doesn't directly provide in https://en.cppreference.com/w/cpp/filesystem/path/append

Reply:
Yes, it's bad wording on my side. Thanks for the suggestion.